src/memory.rs

```diff
@@ -0,0 +1,53 @@
+use crate::constants::MEMORY_MAX;
+
+pub struct Memory {
+    memory: [u16; MEMORY_MAX as usize],
+}
+
+impl Memory {
+    pub fn read(&self, addr: u16) -> u16 {
+        self.memory.get(addr as usize).unwrap().clone()
+    }
+
+    pub fn write(&mut self, addr: u16, data: u16) {
+        *self.memory.get_mut(addr as usize).unwrap() = data;
+    }
```

Comment:
Using `.get().unwrap()` is effectively the same as using `[]`, and in this case it's guaranteed to succeed*, due to the type of `addr` being `u16` and `memory` being a fixed size array.

*After fixing my previous comment.

Reply:
You're right! I was actually planning on implementing better error management (and unwrapping in the meantime), but I get that by using a `u16` it's impossible to index out of the memory bounds.